src/redux/reducers/siloDomainData/leads.js

```diff
@@ -261,6 +268,31 @@ const patchLead = (state, action) => {
     return update(state, settings);
 };
 
+const removeBulkLead = (state, action) => {
+    const { activeProject } = state;
+    const { leadIds } = action;
+    const { totalLeadsCount } = state.leadPage[activeProject];
+
+    const settings = {
+        leadPage: {
+            [activeProject]: {
+                table: { $auto: {
+                    leads: { $autoArray: {
+                        $filter: ld => !leadIds.includes(ld.id),
+                    } },
+                } },
+                grid: { $auto: {
+                    leads: { $autoArray: {
+                        $filter: ld => !leadIds.includes(ld.id),
+                    } },
+                } },
+                totalLeadsCount: { $set: totalLeadsCount - leadIds.length },
```

Comment:
We can use $apply instead of $set.

Also, the function assumes that leadIds is subset of table.leads and table.grids which may not be the case.